src/domManipulators/copySchedules.ts

```diff
@@ -178,48 +185,55 @@ async function handleCopyScheduleButtonClick(
 
       rowData.push(cellText)
     })
-
     tableData.push(rowData)
+
+    for (const cell of rowCells) {
+      const courseId = extractIdFromDOM(cell)
+      if (courseId) {
+        rowData.push(courseId)
+        break
+      }
+    }
   })
-  const button = document.querySelector(
-    '.NewSectionButton[title="Add Section"]'
-  ) as HTMLElement
 
+  const selectedSections: ISectionData[] = []
+  const currentWorklistNumber = (
+    await chrome.storage.local.get("currentWorklistNumber")
+  ).currentWorklistNumber
   for (let i = 2; i < tableData.length; i++) {
     // Change column that course code is being taken from depending on button type
     const code =
       buttonType === "saved"
         ? tableData[i][3].slice(0, tableData[i][3].indexOf(" - "))
         : tableData[i][4].slice(0, tableData[i][4].indexOf(" - "))
 
-    // The following await-in-loop is not currently parallelizable as
-    // each course is manually loaded in by clicking the NewSectionButton.
-    //
-    // TODO: refactor this to add sections more directly.
-    // eslint-disable-next-line no-await-in-loop
-    const selectedSection = await findCourseInfo(code)
+    let selectedSection = null
+    console.log(tableData[i][tableData[i].length - 1])
+    try {
+      if (tableData[i][tableData[i].length - 1] !== "") {
+        // eslint-disable-next-line no-await-in-loop
+        selectedSection = await fetchWorkdayData(
+          tableData[i][tableData[i].length - 1]
+        )
+      } else {
+        // eslint-disable-next-line no-await-in-loop
+        selectedSection = await findCourseInfo(code)
+      }
+    } catch {
+      continue
```

Comment:
it would be good to store which sections couldn't be added, and alert the user at the end so they're aware of it